gauss_jordan.go

```diff
@@ -2,44 +2,47 @@ package modular
 
 import (
 	"errors"
-	// "fmt"
-	// "math/big"
 )
 
+// Returns the index of the first non-zero entry of the array, along with such value.
+// If no such value exists then return -1 as the index and nil as the value
 func NonZeroEntry(arr []*Int) (int, *Int, error) {
 	if len(arr) == 0 {
 		return 0, nil, errors.New("Empty array is not valid")
 	}
-	nz := arr[0]
-	idx := int(0)
+	value := arr[0]
+	index := -1
 	for i, a := range arr {
 		if a.Cmp(NewInt(0)) != 0 {
-			nz = a
-			idx = i
+			value = a
+			index = i
```

Comment:
idx is fine too ;)

Reply:
I don't mind long names to be honest. If it's fine I would prefer to stick to long names :)